src/PaymentGateways/PayPalCommerce/onBoardingRedirectHandler.php

```diff
@@ -390,53 +391,29 @@ private function isAdminSuccessfullyOnBoarded($merchantId, $accessToken, $usesCu
         }
 
         if (! $onBoardedData['primary_email_confirmed']) {
-            $errorMessage[] = esc_html__('Confirm your primary email address', 'give');
+            $errorMessages[] = esc_html__('Confirm your primary email address', 'give');
         }
 
-        if (! $usesCustomPayments) {
-            return count($errorMessages) > 1 ? $errorMessages : true;
-        }
+        // This error message is only for the case when the user is using custom payments.
+        // Host card fields are supported only for specific countries and PayPal seller account of PPCP type.
+        if( $usesCustomPayments ) {
+            $sellerCapabilities = wp_list_pluck( $onBoardedData['capabilities'], 'name');
+            $requiredCapability = 'CUSTOM_CARD_PROCESSING';
+            $hasCustomCardProcessingCapability = array_search($requiredCapability, $sellerCapabilities, true);
 
-        if (array_diff(['products', 'capabilities'], array_keys($onBoardedData))) {
-            $errorMessages[] = esc_html__(
-                'Your account was expected to be able to accept custom payments, but is not. Please make sure your
-				account country matches the country setting. If the problem persists, please contact PayPal.',
-                'give'
-            );
+            if( false !== $hasCustomCardProcessingCapability ) {
+                /* @var \stdClass $customCardProcessingCapability */
+                $customCardProcessingCapability = $sellerCapabilities[$hasCustomCardProcessingCapability];
 
-            // Return here since the rest of the validations will definitely fail
-            return $errorMessages;
-        }
-
-        // Grab the PPCP_CUSTOM product from the status data
-        $customProduct = current(
-            array_filter(
-                $onBoardedData['products'],
-                function ($product) {
-                    return $product['name'] === 'PPCP_CUSTOM';
+                if( 'ACTIVE' !== $customCardProcessingCapability->status ){
+                    $errorMessages[] = __(
+                        'Custom card processing is not active. This means you can accept payment only with <a href="https://developer.paypal.com/docs/checkout/" title="Link to PayPal Docs">PayPal smart buttons</a>. To accept payment with <a href="https://developer.paypal.com/docs/checkout/advanced/#enable-your-account" title="Link to PayPal Docs">PayPal hosted card field</a>, Please contact PayPal to resolve this issue.',
```

Comment:
@Benunc Please review this notice and Let time know if you want to update it.

cc: @JasonTheAdams